lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -459,28 +457,30 @@ def access_token_url
       end
 
       def setup_access_token
-        request = 'grant_type=client_credentials'
-        begin
-          raw_response = ssl_post(access_token_url, request, access_token_header)
-        rescue ResponseError => e
-          raise OAuthResponseError.new(e)
-        else
-          response = parse(raw_response)
+        response = parse(ssl_post(access_token_url, 'grant_type=client_credentials', access_token_header))
+        @options[:access_token] = response['access_token']
+        @options[:expires] = (DateTime.now + response.fetch('expires_in', 0).seconds).strftime('%Q').to_i
 
-          if (access_token = response['access_token'])
-            access_token
-          else
-            raise OAuthResponseError.new(response)
-          end
-        end
+        Response.new(
+          response['access_token'].present?,
+          message_from(true, response, {}),
+          response,
+          test: test?,
+          error_code: error_code_from(false, response, {})
+        )
+      rescue ResponseError => e
+        Response.new(false, e.response.body.to_s, {}, test: test?)
```

Comment:
Considering the way that y'all wrote this PR think this could be an OAuthResponseError instead

Reply:
Sure, I will

Reply:
done!